src/core/next.ts

```
@@ -33,11 +34,18 @@ function safeEval(code: string, routePath: string) {
-  const content = await fs.readFile(routePath, "utf-8");
-  const code = transpile(content, routeDefinerName);
+  const rawCode = await fs.readFile(routePath, "utf-8");
+  const middlewareName = detectMiddlewareName(rawCode);
+  const code = transpile(rawCode, routeDefinerName, middlewareName);
   const fixedCode = Object.keys(schemas).reduce(injectSchemas, code);
   (global as Record<string, unknown>).schemas = schemas;
+  if (middlewareName) {
+    // (global as Record<string, unknown>)[middlewareName] = () => { /* mock */ };
+  }
   const result = safeEval(fixedCode, routePath);
   delete (global as Record<string, unknown>).schemas;
```

Biome

[error] Avoid the delete operator which can impact performance.

Unsafe fix: Use an undefined assignment instead.


(lint/performance/noDelete)